src/vanilla/store2.ts

```diff
@@ -649,114 +584,31 @@ export const createStore = (): Store => {
-        devListenersRev2.forEach((l) => l({ type: 'unsub' }))
-        devListenersRev3.forEach((l) => l({ type: 'unsub' }))
-      }
     }
   }
 
   if (import.meta.env?.MODE !== 'production') {
-    return {
+    const store: Store = {
       get: readAtom,
       set: writeAtom,
       sub: subscribeAtom,
       // store dev methods (these are tentative and subject to change without notice)
-      dev_subscribe_store: (l) => {
-        devListenersRev2.add(l)
-        return () => {
-          devListenersRev2.delete(l)
-        }
-      },
-      dev_get_mounted_atoms: () => mountedAtoms.values(),
-      dev_get_atom_state: (a: AnyAtom) => {
-        const getOldAtomState = (a: AnyAtom): OldAtomState | undefined => {
-          const aState = atomStateMap.get(a)
-          return (
-            aState &&
-            aState.s && {
-              d: new Map<AnyAtom, OldAtomState>(
-                Array.from(aState.d.keys()).flatMap((a) => {
-                  const s = getOldAtomState(a)
-                  return s ? [[a, s]] : []
-                }),
-              ),
-              ...aState.s,
-            }
-          )
-        }
-        return getOldAtomState(a)
-      },
-      dev_get_mounted: (a: AnyAtom) => {
-        const aState = atomStateMap.get(a)
-        return (
-          aState &&
-          aState.m && {
-            l: aState.m.l,
-            t: new Set([...aState.t, a]), // HACK to include self
-            ...(aState.m.u ? { u: aState.m.u } : {}),
-          }
-        )
-      },
-      dev_restore_atoms: (values: Iterable<readonly [AnyAtom, AnyValue]>) => {
-        const pendingPair = createPendingPair()
-        for (const [atom, value] of values) {
-          setAtomStateValueOrPromise(getAtomState(atom), value)
-          recomputeDependents(pendingPair, atom)
-        }
-        const flushed = flushPending(pendingPair)
-        devListenersRev2.forEach((l) =>
-          l({ type: 'restore', flushed: flushed! }),
-        )
-      },
-      dev3_subscribe_store: (l) => {
-        devListenersRev3.add(l)
-        return () => {
-          devListenersRev3.delete(l)
-        }
-      },
-      dev3_get_mounted_atoms: () => mountedAtoms.values(),
-      dev3_get_atom_state: (a) => atomStateMap.get(a)?.s,
-      dev3_get_atom_deps: (a) => {
-        const aState = atomStateMap.get(a)
-        return aState ? new Set(aState.d.keys()) : undefined
-      },
-      dev3_restore_atoms: (values) => {
-        const pendingPair = createPendingPair()
-        for (const [atom, value] of values) {
-          setAtomStateValueOrPromise(getAtomState(atom), value)
-          recomputeDependents(pendingPair, atom)
-        }
-        const flushed = flushPending(pendingPair)
-        flushed?.forEach((a) => {
-          devListenersRev3.forEach((l) => l({ type: 'set', atom: a }))
-        })
+      dev4_get_internal_weak_map: () => atomStateMap,
+      dev4_override_method: (key, fn) => {
+        ;(store as any)[key] = fn
```

Comment:
Nice! A few questions

1. With this approach users would probably need to `composeWithDevTools(store || getDefaultStore())` and supply it via `<Provider/>` to their application. So the DevTools would no longer support the provider-less mode unless if we keep `composeWithDevTools` function within the main lib. Any thoughts around this? 🤔 

2. What would be the best way to wait for atoms resolving a promise (in DevTools)? i.e. values changing from sync -> async or async -> async (new promise). This did not exist in v1 btw, but would be nice to 
include it in v2

3. Could we still keep `mountedAtoms` around? Perhaps we could introduce `dev4_get_internal_mounted_atom_map`? If you feel strongly about not including it, then I could start with keeping references to the atoms changed with `subscribe/unsubscribe` 🤔  

4. Do we still need `dev4` prefix or should we call it `dev_get_internal_weak_map` etc.? From Jotai DevTools perspective, exposing `import { DevTools } from 'jotai-devtools/experimental` might make more sense that eliminates any conflict with the `dev` vs `dev4`

Reply:
> 1. With this approach users would probably need to `composeWithDevTools(store || getDefaultStore())` and supply it via `<Provider/>` to their application. So the DevTools would no longer support the provider-less mode unless if we keep `composeWithDevTools` function within the main lib. Any thoughts around this?

I thought without `dev4_override_method`, that's the case, and hoped `dev4_override_method` could work it around.

Reply:
> 2\. What would be the best way to wait for atoms resolving a promise (in DevTools)?

I assume you can simply `await` the promise value from `store.get()`.

Reply:
> 3\. Could we still keep `mountedAtoms` around?

Hmm, that was one of the biggest things, I would like to avoid in dev4. Otherwise, we can re-consider dev3.

Reply:
> 4\. Do we still need `dev4` prefix or

I don't understand what conflicts. I prefer the revision number for clearer communication. (we may have `dev5` in the future.)

Reply:
> I thought without dev4_override_method, that's the case, and hoped dev4_override_method could work it around.

Since it is updating the reference of the store it should work with the default store too 🤔

> Hmm, that was one of the biggest things, I would like to avoid in dev4. Otherwise, we can re-consider dev3.

I can work around it by patching `store.set` and manually keeping track of the atoms by sub and unsub. Not sure if I'm missing any edge cases here.

> I don't understand what conflicts. 

Mostly TypeScript related, I'm trying to avoid the number of type guards 😅 but I can circumvent it.